src/tasks/AotCompilerTask/MonoAOTCompiler.csproj

```
@@ -16,6 +19,7 @@
   <ItemGroup>
     <Compile Include="MonoAOTCompiler.cs" />
     <Compile Include="..\Common\Utils.cs" />
+    <Compile Include="$(RepoRoot)src\libraries\System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/NullableAttributes.cs" Condition="$(TargetFramework.StartsWith('net4'))" />
```

In item evaluation, the `TargetFramework` has already been parsed into `TargetFrameworkIdentifier` and other properties, so you can use them in conditions.

```suggestion
    <Compile Include="$(RepoRoot)src\libraries\System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/NullableAttributes.cs" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'" />
```